diff --git a/python-package/setup.py b/python-package/setup.py
index d4b6aa6..a81adfa 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -26,6 +26,12 @@ for libfile in libpath['find_lib_path']():
         continue

 print("Install libxgboost from: %s" % LIB_PATH)
+if os.getenv('CONDA_BUILD'):
+    print("Conda-build detected, not installing (xgboost nor) libxgboost from: %s" % LIB_PATH)
+    data_files=('xgboost')
+else:
+    print("Install libxgboost from: %s" % LIB_PATH)
+    data_files=('xgboost', LIB_PATH)
 # Please use setup_pip.py for generating and deploying pip installation
 # detailed instruction in setup_pip.py
 setup(name='xgboost',
@@ -43,7 +49,7 @@ setup(name='xgboost',
       # this will use MANIFEST.in during install where we specify additional files,
       # this is the golden line
       include_package_data=True,
-      data_files=[('xgboost', LIB_PATH)],
+      data_files=[data_files],
       license='Apache-2.0',
       classifiers=['License :: OSI Approved :: Apache Software License'],
       url='https://github.com/dmlc/xgboost')
diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
index d87922c..8158b5a 100644
--- a/python-package/xgboost/libpath.py
+++ b/python-package/xgboost/libpath.py
@@ -24,6 +24,10 @@ def find_lib_path():
     dll_path = [curr_path, os.path.join(curr_path, '../../lib/'),
                 os.path.join(curr_path, './lib/'),
                 os.path.join(sys.prefix, 'xgboost')]
+    if sys.platform == 'win32':
+        dll_path = [os.path.join(sys.prefix, 'Library', 'mingw-w64', 'bin')]
+    else:
+        dll_path = [os.path.join(sys.prefix, 'lib')]
     if sys.platform == 'win32':
         if platform.architecture()[0] == '64bit':
             dll_path.append(os.path.join(curr_path, '../../windows/x64/Release/'))
