diff --git a/R-package/src/Makevars.in b/R-package/src/Makevars.in
index 7a1d06d..5f3641d 100644
--- a/R-package/src/Makevars.in
+++ b/R-package/src/Makevars.in
@@ -1,5 +1,5 @@
 # package root
-PKGROOT=../../
+PKGROOT=../..
 ENABLE_STD_THREAD=1
 # _*_ mode: Makefile; _*_

diff --git a/R-package/src/Makevars.win b/R-package/src/Makevars.win
index 2ceeb84..84e7ef5 100644
--- a/R-package/src/Makevars.win
+++ b/R-package/src/Makevars.win
@@ -1,5 +1,5 @@
 # package root
-PKGROOT=./
+PKGROOT=../..
 ENABLE_STD_THREAD=0
 # _*_ mode: Makefile; _*_
