diff --git a/make/mingw64.mk b/make/mingw64.mk
index e20220b..d8659a9 100644
--- a/make/mingw64.mk
+++ b/make/mingw64.mk
@@ -6,8 +6,12 @@
 #
 # see config.mk for template.
 #-----------------------------------------------------------
-export CXX=g++ -m64
-export CC=gcc -m64
+ifndef CC
+export CC=gcc
+endif
+ifndef CXX
+export CXX=g++
+endif

 # Whether enable openmp support, needed for multi-threading.
 USE_OPENMP = 1
