diff --git a/python-package/setup.py b/python-package/setup.py
index 8a1b1b70..658a7cb2 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -342,12 +342,6 @@ if __name__ == '__main__':
           ],
           ext_modules=[CMakeExtension('libxgboost')],
           # error: expected "str": "Type[Command]"
-          cmdclass={
-              'build_ext': BuildExt,     # type: ignore
-              'sdist': Sdist,            # type: ignore
-              'install_lib': InstallLib,  # type: ignore
-              'install': Install          # type: ignore
-          },
           extras_require={
               'pandas': ['pandas'],
               'scikit-learn': ['scikit-learn'],
